GIT 提交信息：ResetPassword

huan5678 opened the pull request:
功能：重構密碼重置流程及視圖

- 在路由索引中增加一條新的 `ResetPassword` 路徑。
- 從路由索引中移除現有的 `ResetPassword` 路徑。
- 實現一個新的 `ResetPassword.vue` 視圖用於密碼重置。
- 移除 `backend/ResetPassword.vue` 視圖及其相關樣式和腳本。
- 更新 `LoginView.vue` 按鈕和第三方登錄鏈接以包含具有不同樣式的禁用狀態。

huan5678 committed:
GIT 提交信息：

功能：重構密碼重置流程及視圖

- 在路由索引中增加一條新的 `ResetPassword` 路徑。
- 從路由索引中移除現有的 `ResetPassword` 路徑。
- 實現一個新的 `ResetPassword.vue` 視圖用於密碼重置。
- 移除 `backend/ResetPassword.vue` 視圖及其相關樣式和腳本。
- 更新 `LoginView.vue` 按鈕和第三方登錄鏈接以包含具有不同樣式的禁用狀態。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

echocarriet commented:
測試登入頁面以帳密登入後台，顯示 success 後，進入空白後台，需重新整理才會顯示後台資料的畫面。

eee317 commented:
@echocarriet 你提到的問題，應該我那邊的 nav 合併後就可以解決了 :)

eee317 commented:
我發現不管是在部署的位置還是這支 PR 或其他 PR ，因為忘記之前註冊的帳號，所以重新新註冊想要看重設密碼這塊
結果我發現手動新註冊都沒有辦法成功

echocarriet commented:
> @echocarriet 你提到的問題，應該我那邊的 nav 合併後就可以解決了 :)

好喔，那等妳那邊合併我再測試看看。

huan5678 commented:
不成功的原因在於 流程錯誤直接進入resetpassword的頁面是不能改的 這個頁面目前是接由重置密碼信件裡面的網址跳轉過來的方式所以預設讀網址列上的token往後端送去做重置密碼的動作，如果本來就沒有token自然是不能重設密碼

huan5678 committed:
GIT 提交信息：

功能：增強密碼重設流程及工具

- 新增一個處理 cookies 的工具。
- 實現密碼重設邏輯，使用 cookie 令牌（如果路由查詢令牌未定義）。
- 在成功使用 cookie 令牌重設密碼後，以本地語言顯示成功訊息並重定向到登入頁面。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
Merge branch 'main' into dev/resetpassword

eee317 commented:
我目前 run 起來會有這個錯誤訊息唷
<img width="838" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/5b98b1cb-27c6-4b20-bf1f-404bd814ffb6">

huan5678 committed:
[fix] login register bug

huan5678 committed:
GIT 提交訊息：

功能：優化認證界面和導航

- 移除 `LoginView.vue` 中，郵件和密碼欄位的 `VField` 組件上的水平間距類別，改用一致的間距類別。
- 移除 `LoginView.vue` 中有關同意隱私政策部分的內容。
- 更新 `SignupView.vue` 中表單提交綁定，從阻止修改器改為正常綁定。
- 將在 `LoginView.vue` 中所做的相同間距類別更新應用於 `SignupView.vue` 的 `VField` 組件。
- 在 `SignupView.vue` 成功註冊後，於 SweetAlert 彈出視窗中增加成功圖標和文字提示。
- 修改 `SignupView.vue` 中註冊成功後的導航，從使用 Vue 路由器導航改為直接改變窗口位置。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

eee317 commented on src/router/index.js:
```diff
@@ -41,6 +41,12 @@ const routes = [
         component: () => import('../views/LoginView.vue'),
       },
+      {
+        dev/resetpassword
```
這行似乎會噴錯 > <
<img width="788" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/f3069840-0ace-4bc3-b01d-b66b59dba4f0">

huan5678 committed:
fix: router.js

eee317 commented:
目前看靜態頁面都沒有問題
剛剛使用手動註冊也可以成功了
可能因為下圖的頁面還沒有出來，比較難有 token 去測試重設密碼是否OK
<img width="592" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/71a980dc-594d-4953-825b-4f114cdb7956">

有試過登入狀態下切換到 /resetpassword 的路徑
但是更換秘密碼後，點送出會卡住
<img width="1253" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/8e69a2e9-3c0b-4842-bc68-8691113cef01">
<img width="1255" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/c43769d5-232c-4623-a3aa-51cc89473ffa">

huan5678 committed:
GIT 提交信息：

修復：精煉密碼重置流程

- 統一密碼及確認密碼輸入欄位的填充
- 從 resetPassword 函數移除 console.log 語句
- 更新 resetPassword 函數中的 put 請求 URL，包含 `/api/auth/reset-password` 路徑

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat: 重構密碼管理流程

- 從路由器路徑移除導向「about」、「privacy」、「terms」、「faq」、「signup」、「login」和「poll」等的開頭斜線
- 在路由器配置中新增帶有元件視圖的「change-password」路徑
- 調整 ResetPassword.vue 中 useCookie 工具的導入路徑，使用別名而非相對路徑
- 當沒有可用的令牌時，移除重設密碼的多餘代碼分支
- 新增更換密碼的視圖，「ChangePasswordView.vue」
- 在新的 ChangePasswordView 元件中實作更改用戶密碼的 UI 和功能
- 透過在成功重設密碼後刪除「selectWaveToken」cookie 來增強安全性

Signed-off-by: 黃士桓 <huan5678@gmail.com>

eee317 commented:
在 admin/change-password 可以進行使用者密碼變更
密碼變更完成後，會被登出，但是上面的 navbar 狀態卻沒有變回未登入狀態
<img width="748" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/dfb7529a-db13-4434-973a-2369ff4a33b0">

huan5678 committed:
GIT 提交信息：

功能：優化用戶介面和身份驗證處理

- 更新 HomeView.vue 中的 UI 佈局和屬性順序，以提高可讀性和標準化
- 刪除用戶身份驗證 cookie 並在頁面加載或更改密碼過程中更新兩個 Vue 組件（ResetPassword.vue 和 ChangePasswordView.vue）的用戶狀態

Signed-off-by: 黃士桓 <huan5678@gmail.com>
